show/main.py

```diff
@@ -1091,12 +1091,42 @@ def syseeprom(verbose):
 @click.option('--verbose', is_flag=True, help="Enable verbose output")
 def psustatus(index, verbose):
     """Show PSU status information"""
-    cmd = "sudo psuutil status"
-
-    if index >= 0:
-        cmd += " -i {}".format(index)
+    db = SonicV2Connector(host="127.0.0.1")
+    db.connect(db.STATE_DB)
+
+    # Currently set chassis_num to 1, need to improve it once new platform API is implemented
+    chassis_num = 1
+    chassis_name = "chassis{}".format(chassis_num)
+    num_psus = db.get(db.STATE_DB, 'CHASSIS_INFO|{}'.format(chassis_name), 'num_psus')
+    if not num_psus:
+        click.echo("Error! PSU daemon is broken!")
+        return
+    supported_psu = range(1, int(num_psus) + 1)
+    if (index < 0):
+        psu_ids = supported_psu
+    else:
+        psu_ids = [index]
+
+    header = ['PSU', 'Status']
+    status_table = []
+
+    for psu in psu_ids:
+        msg = ""
+        psu_name = "PSU{}".format(psu)
+        if psu not in supported_psu:
```

Comment:
I would move this validation to 1108
implementation looks better when it validates input first and then processes it

Reply:
oh, if what we get in num_psus is not equal to "actually installed"
message should be changed to something like "PSUX is not installed"

Reply:
The printing message here is just copied from psuutil, so shall we just keep as it is?

Reply:
ok, just noticed user can specify the PSU index.
in such case message makes sense.

Reply:
Yes, thanks for the comments all the same. :)